File: spec/php/ProcessRotateTest.php
<?php
namespace Kaitai\Struct\Tests;

class ProcessRotateTest extends TestCase {
    public function testProcessRotate() {
        $r = ProcessRotate::fromFile(self::SRC_DIR_PATH . "/process_rotate.bin");

        $this->assertEquals("Hello", $r->buf1());
        $this->assertEquals("World", $r->buf2());
        $this->assertEquals("There", $r->buf3());
    }
}

File: spec/go/process_rotate_test.go
package spec

import (
	"os"
	"testing"

	"github.com/kaitai-io/kaitai_struct_go_runtime/kaitai"
	"github.com/stretchr/testify/assert"

	. "test_formats"
)

func TestProcessRotate(t *testing.T) {
	f, err := os.Open("../../src/process_rotate.bin")
	if err != nil {
		t.Fatal(err)
	}
	s := kaitai.NewStream(f)

	var r ProcessRotate
	err = r.Read(s, &r, &r)
	if err != nil {
		t.Fatal(err)
	}

	assert.EqualValues(t, "Hello", r.Buf1)
	assert.EqualValues(t, "World", r.Buf2)
	assert.EqualValues(t, "There", r.Buf3)
}
